components/style/stylist.rs

```diff
@@ -98,6 +101,16 @@ pub struct Stylist {
 
     /// The total number of times the stylist has been rebuilt.
     num_rebuilds: usize,
+
+    /// The set of registered custom property associated with the document.
+    /// Should be initialized as soon as possible.
+    #[cfg_attr(feature = "servo", ignore_heap_size_of = "Arc")]
+    registered_property_set: Option<Arc<RwLock<RegisteredPropertySet>>>,
```

Comment:
As noted, the Stylo CSS.registerProperty/unregisterProperty is in a separate patch. That goes through the same codepath (properties_and_values::register_property) as Servo's CSS.registerProperty/unregisterProperty.

Going to try using the SharedRwLock!